### PR TITLE
276 anonymous session

### LINKED ISSUE #276: [sdk-middleware-auth] Implement anonymous session auth flow
### Description

Require implementation of session flow for `/oauth/key/anonymous/token/` endpoint.

### PR DESCRIPTION
**Pull request by hisabimbola**
#### Summary

Fixes #276 
#### Todo

- Tests
    - [ ] Unit
    - [ ] Integration
    - [ ] Acceptance
- [ ] Documentation

### CODE REVIEW

**Comment by hisabimbola** [comment]
This PR was built from #281, so make sense to review after that PR has been merged.

File: packages/sdk-middleware-auth/test/build-requests.spec.js [hunk 1]
```diff
@@ -15,12 +17,131 @@ function createTestOptions (options) {
...
+    ).toThrowError(
+      'Missing required credentials (clientId, clientSecret, user)',
+    )
+  })
+
+  it('validate required option (username, password)', () => {
+    const options = createTestOptions({
+      credentials: {
+        clientId: 'yeah',
+        clientSecret: 'yo',
```
Comment (adnasa) [inline packages/sdk-middleware-auth/test/build-requests.spec.js]: - clientId: foo
- clientSecret: bar

File: packages/sdk-middleware-auth/test/build-requests.spec.js [hunk 2]
```diff
@@ -110,3 +231,31 @@ describe('buildRequestForClientCredentialsFlow', () => {
...
+      url: 'http://localhost:8080/oauth/test/anonymous/token',
+      body: `grant_type=client_credentials&scope=${allScopes.join(' ')}`,
+    })
+  })
+
+  it('should add anonymousId if passed in', () => {
+    const mockCred = {
+      clientId: '123',
+      clientSecret: 'secret',
+      anonymousId: 'youdontknowme',
```
Comment (adnasa) [inline packages/sdk-middleware-auth/test/build-requests.spec.js]: - clientId: foo
- clientSecret: bar
- anonymousId: baz

File: packages/sdk-middleware-auth/test/build-requests.spec.js [hunk 3]
```diff
@@ -110,3 +231,31 @@ describe('buildRequestForClientCredentialsFlow', () => {
...
+    }
+    const options = createTestOptions({ credentials: mockCred })
+    expect(buildRequestForAnonymousSessionFlow(options)).toEqual({
+      basicAuth: 'MTIzOnNlY3JldA==',
+      url: 'http://localhost:8080/oauth/test/anonymous/token',
+      body: oneLineTrim`grant_type=client_credentials&
```
Comment (adnasa) [inline packages/sdk-middleware-auth/test/build-requests.spec.js]: use new lines to increase readability here?

File: packages/sdk-middleware-auth/test/password-flow.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      const next = (actualParams) => {
+        expect(actualParams.request).toEqual(request)
+        expect(actualParams.response).toEqual(response)
+        expect(actualParams.pendingTasks).toEqual([])
+        expect(actualParams.url).toBe(
+          'https://auth.commercetools.co/oauth/foo/customers/token',
+        )
+        expect(actualParams.basicAuth).toBe('MTIzOnNlY3JldA==')
+        expect(authMiddlewareBase).toHaveBeenCalledTimes(1)
+        jest.unmock('../src/base-auth-flow')
```
Comment (adnasa) [inline packages/sdk-middleware-auth/test/password-flow.spec.js]: shouldn't this be in an `afterEach` or `afterAll`?

File: packages/sdk-middleware-auth/test/password-flow.spec.js [hunk 5]
```diff
@@ -4,6 +4,9 @@ import {
 
 import authMiddlewareBase from '../src/base-auth-flow'
 
+// required to be at the root because Jest hoists it avoid all requires,
```
Comment (wizzy25) [inline packages/sdk-middleware-auth/test/password-flow.spec.js]: Do you mean `above all requires`?

**Pull request merged**